src/psyclone/psyir/transformations/arrayassignment2loops_trans.py

```diff
@@ -241,17 +241,16 @@ def validate(self, node, options=None):
                 f"Range, but none were found in '{node.debug_string()}'."))
 
         # All the ArrayMixins must have the same number of Ranges to expand
-        found = None
+        num_of_ranges = None
         for accessor in node.walk(ArrayMixin):
-            num_of_ranges = len([x for x in accessor.indices
-                                 if isinstance(x, Range)])
-            if num_of_ranges > 0:
-                if not found:
+            count = len([x for x in accessor.indices if isinstance(x, Range)])
+            if count:
```

Comment:
I like the name changes, but for clarity I suggest still doing "if count > 0", even if technically it is the same.

Reply:
Done.